ftw/trash/patches.py

```diff
@@ -50,3 +51,21 @@ def searchResults(self, REQUEST=None, **kw):
         kw['trashed'] = [True, False, None]
 
     return self._old_searchResults(REQUEST, **kw)
+
+
+def _getFieldObjects(self, *args, **kwargs):
+    return filter(lambda obj: not ITrashed.providedBy(obj),
+                  self._old__getFieldObjects(*args, **kwargs))
+
+
+def getRawActionAdapter(self, *args, **kwargs):
+    adapter_ids = self._old_getRawActionAdapter(*args, **kwargs)
+    adapter_result = []
+
+    for adapter_id in adapter_ids:
+        adapter = getattr(self.aq_explicit, adapter_id, None)
```

Comment:
I would remove the `.aq_explicit` here, it is not necessary for our use case.
We also do not use `aq_*`-attributes, but rather import a function from `Acquisition` instead.
But this is really not necessary here 😉

Reply:
Done